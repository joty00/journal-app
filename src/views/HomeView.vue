<template>
  <div class="d-flex text-success">
    <h1>Welcome to DayBook</h1>
    <button class="btn btn-primary" @click="goToDayBook">Let's Start</button>
  </div>
  
</template>

<script>
  export default {
    methods: {
      goToDayBook() {
        this.$router.push({name: 'no-entry'})
      }
    }
  }
</script>
