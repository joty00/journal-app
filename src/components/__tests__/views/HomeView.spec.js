import { describe, expect, test, vi } from 'vitest'
import HomeView from '@/views/HomeView.vue'
import { shallowMount } from '@vue/test-utils'

describe('Pruebas en el Home View', () => {
    test('debe renderizar el componente correctamente ', () => {
        const wrapper = shallowMount(HomeView)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Hacer click en un boton debe redireccionar a no-entry', () => {

        const mockRouter = {
            push: vi.fn()
        }

        const wrapper = shallowMount(HomeView, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
    })

})