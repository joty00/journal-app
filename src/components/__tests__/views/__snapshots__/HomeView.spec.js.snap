// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en el Home View > debe renderizar el componente correctamente  1`] = `
"<div class=\\"d-flex text-success\\">
  <h1>Welcome to DayBook</h1><button class=\\"btn btn-primary\\">Let's Start</button>
</div>"
`;
